app/src/main/java/fr/free/nrw/commons/explore/depictions/SearchDepictionsFragment.java

```diff
@@ -0,0 +1,238 @@
+import butterknife.ButterKnife;
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.depictions.DepictedImagesActivity;
+import fr.free.nrw.commons.di.CommonsDaggerSupportFragment;
+import fr.free.nrw.commons.explore.recentsearches.RecentSearch;
+import fr.free.nrw.commons.explore.recentsearches.RecentSearchesDao;
+import fr.free.nrw.commons.kvstore.JsonKvStore;
+import fr.free.nrw.commons.mwapi.MediaWikiApi;
+import fr.free.nrw.commons.upload.structure.depicts.DepictedItem;
+import fr.free.nrw.commons.utils.NetworkUtils;
+import fr.free.nrw.commons.utils.ViewUtil;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+
+public class SearchDepictionsFragment extends CommonsDaggerSupportFragment {
+    private static int TIMEOUT_SECONDS = 15;
+
+    @BindView(R.id.imagesListBox)
+    RecyclerView depictionsRecyclerView;
+    @BindView(R.id.imageSearchInProgress)
+    ProgressBar progressBar;
+    @BindView(R.id.imagesNotFound)
+    TextView depictionNotFound;
+    String query;
+    @BindView(R.id.bottomProgressBar)
+    ProgressBar bottomProgressBar;
+    boolean isLoadingDepictions;
+    @Inject
+    RecentSearchesDao recentSearchesDao;
+    @Inject
+    MediaWikiApi mwApi;
+    @Inject
+    DepictsClient depictsClient;
+    @Inject
+    @Named("default_preferences")
+    JsonKvStore basicKvStore;
+    private RVRendererAdapter<DepictedItem> depictionsAdapter;
+    private List<DepictedItem> queryList = new ArrayList<>();
+
+    private final SearchDepictionsAdapterFactory adapterFactory = new SearchDepictionsAdapterFactory(item -> {
+        // Called on Click of a individual depicted Item
+        // Open Depiction Details activity
+        DepictedImagesActivity.startYourself(getContext(), item);
+        saveQuery(query);
+    });
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_browse_image, container, false);
+        ButterKnife.bind(this, rootView);
+        if (getActivity().getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
+            depictionsRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+        } else {
+            depictionsRecyclerView.setLayoutManager(new GridLayoutManager(getContext(), 2));
+        }
+        ArrayList<DepictedItem> items = new ArrayList<>();
+        depictionsAdapter = adapterFactory.create(items);
+        depictionsRecyclerView.setAdapter(depictionsAdapter);
+        depictionsRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+                super.onScrollStateChanged(recyclerView, newState);
```

Comment:
Is this an attempt to paginate the Recycler View, if yes, this is not how it should ideally be done. Please refer [this](https://medium.com/@etiennelawlor/pagination-with-recyclerview-1cb7e66a502b), or any other reference on how to paginate the recycler view